docs/api_specification.md

```diff
@@ -0,0 +1,159 @@
+| ------------------------ |:--------:| ---------------------------------------------------------------- |----------------------|
+| `external_service_id`    |    X     | external service id of the Service as identified by adminusers.  |   |
+| `pay_api_token`          |    X     | valid api token for the gateway account of above service which this product takes payments for |  |
+| `description`            |    X     | Description of the product. This will be passed as the `description` when creating the charge | |
+| `price`                  |    X     | Price for the product in pence. This will be passed as the  `amount` when creating charge    | |
+| `return_url`             |          | Where to redirect to upon completion of a payment. If not provided, `pay-apps` will generate a default url to itself when creating a charge | |
+| `app_external_id`        |          | (optional) external id of the app the product should be linked to. if not provided, this will generate a `System Generated` app and link the product to it | |
+
+### Response example
+
+```
+201 OK
+Content-Type: application/json
+{
+    "external_product_id": "874h5c87834659q345698495",
+    "external_service_id" :  "d76tt83563thcert8urihgiuer",
+    "description":         "Description of the product",
+    "price":               1050,
+    "return_url" :         "https://some.valid.url/"
+    "links": [
+    {
+        "href": "https://govukpay-apps.cloudapps.digital/v1/api/products/874h5c87834659q345698495",
+        "rel" : "self",
+        "method" : "GET"
+    },
+    {
+         "href": "https://govukpay-apps-ui.cloudapps.digital/pay/874h5c87834659q345698495",
+         "rel" : "pay",
+         "method" : "POST"
+    }]
+}
+```
+
+#### Response field description
+
+| Field                    | always present | Description                                   |
+| ------------------------ |:--------------:| --------------------------------------------- |
+| `external_product_id`    | X              | external id of the new product                |
+| `external_service_id`    | X              | external service id of the Service as identified by adminusers.  |   |
+| `description`            | X              | Description of the product |
+| `price`                  | X              | Price for the product in pence      |
+| `return_url`             |                | return url provided. _(not be available if it was not provided)_   |
+| `links.self`             | X              | self GET link to the product. |
+| `links.pay`              | X              | The link in `pay-apps-ui` where a charge for this product will be generated and redirected to GOV.UK Pay |
+
+---------------------------------------------------------------------------------------------------------------------------------------------------------
+
+## GET /v1/api/products/{productId}
```

Comment:
Would there be a need for a list too?

Reply:
👍 yes .. for the list of product urls tab in selfservice. 
Will add one